src/hotspot/cpu/x86/x86.ad

```diff
@@ -8407,6 +8408,57 @@ instruct vmaskcast(vec dst) %{
   ins_pipe(empty);
 %}
 
+instruct vmaskcast_avx(vec dst, vec src) %{
+  predicate(Matcher::vector_length(n) == Matcher::vector_length(n->in(1)) &&
+            Matcher::vector_length_in_bytes(n) != Matcher::vector_length_in_bytes(n->in(1)) &&
+            (UseAVX >= 2 ||
```

Comment:
Pattern looks relevant only for UseAVX=2

Reply:
Do you mean the mask is using k register when `UseAVX > 2` ?

Reply:
UseAVX > 2 signifies AVX512 target where carrier of all currently valid mask IR nodes is an opmask register.

Reply:
I see, thanks! I will change this later.

Reply:
For AVX3, a mask register is used if the vector species is 512-bit wide or the machine supports AVX512VL, so this code is still relevant in this case.

Reply:
I verified that new instruction sequence with vector as mask carrier over AVX2 is performant specially for down casting operations, since new sequence is using vector as a mask carrier so will be valid if corresponding 512 bit vector species is supported by the target i.e. it clears the constraints imposed by match_rule_supported vector.